### PR TITLE
PP-7668 URL Restructure - Transactions

### PR DESCRIPTION
**Pull request by iqbalgds**
With @SandorArpa

- Update to use the new '/account/<external-id>' format for transaction pages.
  - Update controllers.
  - Update templates.
  - Update functional tests.
  - Update UI template tests

### CODE REVIEW

File: app/controllers/transactions/transaction-detail-redirect.controller.js [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+module.exports = async function redirectToTransactionDetail (req, res) {
   const chargeId = req.params.chargeId
   try {
     let charge = await Ledger.transactionWithAccountOverride(chargeId)
     if (userServicesContainsGatewayAccount(charge.gateway_account_id, req.user)) {
       req.gateway_account.currentGatewayAccountId = charge.gateway_account_id
       req.session = { ...req.session, backLink: req.header('Referer') }
       charge = null
-      res.redirect(302, router.generateRoute(paths.transactions.detail, { chargeId }))
+      res.redirect(302, router.generateRoute(formatAccountPathsFor(router.paths.account.transactions.detail, req.account.external_id), { chargeId }))
```
Comment (sfount) [inline app/controllers/transactions/transaction-detail-redirect.controller.js:20]: No need for `router.generateRoute` here, the format account paths works generically similar to the original formatter, it should be possible to:

```suggestion
      res.redirect(302, formatAccountPathsFor(router.paths.account.transactions.detail, req.account.external_id, chargeId))
```

A little bit less to understand and potentially then able to remove the dependency on router. I think this is true of all of the formatters in this PR.

File: app/views/transaction-detail/_refund.njk [hunk 2]
```diff
@@ -1,5 +1,4 @@
-
-<form id="refundForm" action="/transactions/{{charge_id}}/refund" method="post" class="target-to-show {% if flash.genericError %}active{% endif %}">
+<form id="refundForm" action="{{refundUrl}}" method="post" class="target-to-show {% if flash.genericError %}active{% endif %}">
```
Comment (sfount) [inline app/views/transaction-detail/_refund.njk:1]: `formatAccountPathsFor` is available to the template, potentially using it here would reduce the number of things needed to be passed in to the template 

```suggestion
<form id="refundForm" action="{{ formatAccountPathsFor(routes.account.transactions.refund, currentGatewayAccount.external_id, charge_id) }}" method="post" class="target-to-show {% if flash.genericError %}active{% endif %}">
```

**Review by sfount — COMMENTED** [review]
Code looks good, a few quick URL formatting things I noticed; depending on others opinions I don't think they'd need to be blocking

File: app/controllers/transactions/transaction-detail-redirect.controller.js (hunk 1 above)
Comment (iqbalgds) [reply app/controllers/transactions/transaction-detail-redirect.controller.js]: Good spot - looks cleaner.
Done.

File: app/views/transaction-detail/_refund.njk (hunk 2 above)
Comment (iqbalgds) [reply app/views/transaction-detail/_refund.njk]: Good spot - done.

**Comment by iqbalgds** [comment]
Good spots - ready for re-review.

**Pull request closed without merging**